drivers/ioexpander/icjx.c

```diff
@@ -463,15 +487,80 @@ static int icjx_option(FAR struct ioexpander_dev_s *dev, uint8_t pin,
                        int opt, FAR void *value)
 {
   /* TODO: Implementation of iC-JX options should be here. This includes
-   * setup of filters, ADC, interrupts etc. The right way to implement
+   * setup of filters, ADC etc. The right way to implement
    * this would probably be to introduce config structure to
    * include/nuttx/ioexpanders/icjx.h that the user could use for the
    * nibbles configuration.
+   *
+   * Currently only interrupts are implemented.
    */
 
-  gpiowarn("ERROR: iC-JX options are not yet implemted!\n");
+  FAR struct icjx_dev_s *priv = (FAR struct icjx_dev_s *)dev;
+  uint8_t data;
+  uint8_t reg;
+  int ret;
+
+  DEBUGASSERT(priv != NULL && priv->config != NULL);
+
+  gpioinfo("Expander id=%02x pin=%u option=%u\n",
+           priv->config->id, pin, opt);
+
+  if (opt == IOEXPANDER_OPTION_INTCFG)
+    {
+      unsigned int ival = (unsigned int)((uintptr_t)value);
+      ret = nxmutex_lock(&priv->lock);
+      if (ret < 0)
+        {
+          return ret;
+        }
+
+      switch (ival)
+        {
+          case IOEXPANDER_VAL_HIGH:
+          case IOEXPANDER_VAL_LOW:
+          case IOEXPANDER_VAL_RISING:
+          case IOEXPANDER_VAL_FALLING:
```

Comment:
Isn't IOEXPANDER_VAL_BOTH supported?

Reply:
Good catch, I've missed that one. Updated.